src/app/services/myinfo/myinfo.util.ts

```diff
@@ -0,0 +1,205 @@
+import {
+  CATEGORICAL_DATA_DICT,
+  IPersonBasic,
+  MyInfoSource,
+} from '@opengovsg/myinfo-gov-client'
+import bcrypt from 'bcrypt'
+import { StatusCodes } from 'http-status-codes'
+import { get } from 'lodash'
+import moment from 'moment'
+
+import { createLoggerWithLabel } from '../../../config/logger'
+import {
+  BasicField,
+  IFieldSchema,
+  MapRouteError,
+  MyInfoAttribute,
+} from '../../../types'
+import {
+  DatabaseError,
+  MissingFeatureError,
+} from '../../modules/core/core.errors'
+import { ProcessedFieldResponse } from '../../modules/submission/submission.types'
+
+import {
+  HashDidNotMatchError,
+  HashingError,
+  MissingHashError,
+} from './myinfo.errors'
+import { formatAddress, formatPhoneNumber } from './myinfo.format'
+import {
+  IPossiblyPrefilledField,
+  MyInfoHashPromises,
+  VisibleMyInfoResponse,
+} from './myinfo.types'
+
+const logger = createLoggerWithLabel(module)
+const HASH_SALT_ROUNDS = 10
+
+/**
+ * Retrieves the full MyInfo field value. If field is categorical, the
+ * returned value is retrieved from MyInfo data dictionary. Otherwise, it is
+ * parsed accordingly to construct the returned string.
+ * @param myInfoAttr The MyInfo attribute
+ * @param myInfoData MyInfo person data object returned from MyInfoGovAuthClient
+ * @return The full formatted values, if it exists, otherwise the myInfoKey is returned.
+ */
+export const getMyInfoValue = (
+  myInfoAttr: MyInfoAttribute,
+  myInfoData: IPersonBasic,
+): string | undefined => {
+  switch (myInfoAttr) {
+    // Phone numbers
+    case MyInfoAttribute.MobileNo:
+    case MyInfoAttribute.HomeNo:
+      return formatPhoneNumber(myInfoData[myInfoAttr])
+    case MyInfoAttribute.RegisteredAddress:
+    case MyInfoAttribute.BillingAddress:
+    case MyInfoAttribute.MailingAddress:
+      return formatAddress(myInfoData[myInfoAttr])
+    default:
+      // Categorical data lookup
+      if (CATEGORICAL_DATA_DICT[myInfoAttr]) {
+        return get(
+          CATEGORICAL_DATA_DICT,
+          [myInfoAttr, myInfoData[myInfoAttr].value!, 'description'],
+          '',
```

Comment:
there hasn't been any change in business logic here, right?

Reply:
I've rearranged the logic in a (hopefully) cleaner way, but it shouldn't change the behaviour. the old code was this:
```
    const attr = myInfoData[myInfoAttr]
    // Categorical data lookup	
    if (CATEGORICAL_DATA_DICT[myInfoAttr]) {	
      return _.get(	
        CATEGORICAL_DATA_DICT,	
        [myInfoAttr, attr.value, 'description'],	
        '',	
      )	
    }	

    // Not in category, return by case	
    switch (myInfoAttr) {	
      // Phone numbers	
      case 'mobileno':	
      case 'homeno':	
        return this.formatPhoneNumber(attr)	
      case 'regadd':	
      case 'billadd':	
      case 'mailadd':	
        return this.formatAddress(attr)	
      default:	
        return _.get(attr, 'value', '')	
    }
```